variable.c

```diff
@@ -1554,6 +1554,7 @@ rb_ivar_delete(VALUE obj, ID id, VALUE undef)
             }
             else {
                 val = rb_ivar_delete(fields_obj, id, undef);
+                RBASIC_SET_SHAPE_ID(obj, RBASIC_SHAPE_ID(fields_obj));
```

Comment:
This was a bug, in some cases the IMEMO/fields_obj shape_id wasn't forwarded back to the T_CLASS.

But here too it's unclear what that means for namespaces.